src/index.ts

```diff
@@ -175,24 +162,25 @@ export class Authenticator {
 
   /**
    * Extract value of the authentication token from the request cookies.
-   * @param  {Array}  cookies Request cookies.
+   * @param  {Array}  cookieHeaders 'Cookie' request headers.
    * @return {String} Extracted access token. Throw if not found.
    */
-  _getIdTokenFromCookie(cookies) {
-    this._logger.debug({ msg: 'Extracting authentication token from request cookie', cookies });
-    // eslint-disable-next-line no-useless-escape
-    const regex = new RegExp(`${this._userPoolAppId}\..+?\.idToken=(.*?)(?:;|$)`);
-    if (cookies) {
-      for (let i = 0; i < cookies.length; i++) {
-        const matches = cookies[i].value.match(regex);
-        if (matches && matches.length > 1) {
-          this._logger.debug({ msg: '  Found token in cookie', token: matches[1] });
-          return matches[1];
-        }
-      }
+  _getIdTokenFromCookie(cookieHeaders: Array<{ key?: string | undefined, value: string }>) {
+    this._logger.debug({ msg: 'Extracting authentication token from request cookie', cookieHeaders });
+
+    const tokenCookieNamePrefix = `${this._cookieBase}.`;
+    const tokenCookieNamePostfix = '.idToken';
+
+    const cookies = cookieHeaders.flatMap(h => Cookies.parse(h.value));
+    const token = cookies.find(c => c.name.startsWith(tokenCookieNamePrefix) && c.name.endsWith(tokenCookieNamePostfix))?.value;
+
+    if (!token) {
+      this._logger.debug("  idToken wasn't present in request cookies");
+      throw new Error("Id token isn't present in the request cookies");
```

Comment:
also tiny nit that was here before: consistency with "idToken" above or "Id token" here